include/swift/AST/Attr.h

```diff
@@ -1503,6 +1503,8 @@ class DifferentiableAttr final
   void setVJPFunction(FuncDecl *decl) { VJPFunction = decl; }
 
   bool parametersMatch(const DifferentiableAttr &other) const {
+    assert(ParameterIndices);
+    assert(other.ParameterIndices);
```

Comment:
Merge into one line?
```suggestion
    assert(ParameterIndices && other.ParameterIndices);
```